portal-web/test/functional/com/liferay/portalweb/macros/MultiFactorAuthentication.macro

```diff
@@ -228,6 +235,39 @@ definition {
 			value1 = "Your one-time password has been sent by email.");
 	}
 
+	macro verifyCustomOTP {
+		if (isSet(specificURL)) {
+			Navigator.openSpecificURL(url = "${specificURL}");
+		}
+		else {
+			Navigator.openURL();
+		}
+
+		SignIn._signIn(
+			userEmailAddress = "${userEmailAddress}",
+			userPassword = "${userPassword}");
+
+		AssertElementPresent(locator1 = "MultiFactorAuthentication#CUSTOM_MFA_TITLE");
+
+		AssertElementPresent(locator1 = "MultiFactorAuthentication#CUSTOM_MFA_DESCRIPTION");
+
+		AssertElementPresent(locator1 = "MultiFactorAuthentication#EMAIL_OTP_LINK");
+
+		Type(
+			locator1 = "MultiFactorAuthentication#CUSTOM_MFA_VERIFY_FIELD",
+			value1 = "${verificationText}");
+
+		Button.click(button = "Submit");
+
+		if (IsElementPresent(locator1 = "MultiFactorAuthentication#EMAIL_OTP_VERIFY_FAILED_MESSAGE")) {
+			AssertElementNotPresent(locator1 = "UserBar#USER_AVATAR_IMAGE");
+		}
+		else {
+			AssertElementNotPresent(locator1 = "MultiFactorAuthentication#EMAIL_OTP_VERIFY_FIELD");
+			AssertElementPresent(locator1 = "UserBar#USER_AVATAR_IMAGE");
+		}
```

Comment:
Try not to rely on assertions that rely on elements on the screen. This has gone wrong many times in the past.

```
if(${failedVerification} == true){
   Add assertions here;
}
else {
  Happy path assertions here;
}

Reply:
I don't think we need this complex structure for this test case.
@dellawang simplified this assertion in https://github.com/brianchandotcom/liferay-portal/pull/96309, and I use the same for verifyCustomOTP macro.